clients/cli/src/wrap.rs

```diff
@@ -0,0 +1,224 @@
+
+    /// The address of the escrow account that will hold the unwrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub escrow_account: Pubkey,
+
+    /// The address of the token program that the wrapped mint should belong to
+    #[clap(value_parser = parse_token_program)]
+    pub wrapped_token_program: Pubkey,
+
+    /// The amount of tokens to wrap
+    #[clap(value_parser)]
+    pub amount: u64,
+
+    /// Path to the signer for the transfer authority if different from
+    /// fee payer
+    #[clap(long = "transfer-authority", value_name = "PATH")]
+    pub transfer_authority: Option<String>,
+
+    /// The address of the mint to wrap, queried if not provided
+    #[clap(long = "unwrapped-mint", value_parser = parse_pubkey)]
+    pub unwrapped_mint: Option<Pubkey>,
+
+    /// The address of the token account to receive wrapped tokens.
+    /// If not provided, defaults to fee payer associated token account
+    #[clap(long = "recipient-token-account", value_parser = parse_pubkey)]
+    pub recipient_token_account: Option<Pubkey>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WrapOutput {
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_mint_address: Pubkey,
+    #[serde_as(as = "DisplayFromStr")]
+    pub wrapped_mint_address: Pubkey,
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_token_account: Pubkey,
+    #[serde_as(as = "DisplayFromStr")]
+    pub recipient_token_account: Pubkey,
+    #[serde_as(as = "DisplayFromStr")]
+    pub escrow_account: Pubkey,
+    pub amount: u64,
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub signature: Option<Signature>,
+}
+
+impl Display for WrapOutput {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln_name_value(
+            f,
+            "Unwrapped mint address:",
+            &self.unwrapped_mint_address.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Wrapped mint address:",
+            &self.wrapped_mint_address.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Unwrapped token account:",
+            &self.unwrapped_token_account.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Recipient wrapped token account:",
+            &self.recipient_token_account.to_string(),
+        )?;
+        writeln_name_value(f, "Escrow account:", &self.escrow_account.to_string())?;
+        writeln_name_value(f, "Amount:", &self.amount.to_string())?;
+        if let Some(signature) = self.signature {
+            writeln_name_value(f, "Signature:", &signature.to_string())?;
+        }
+        Ok(())
+    }
+}
+
+impl QuietDisplay for WrapOutput {
+    fn write_str(&self, _: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        Ok(())
+    }
+}
+impl VerboseDisplay for WrapOutput {}
+
+async fn get_unwrapped_mint(
+    rpc_client: &Arc<RpcClient>,
```

Comment:
nit: passing in a `&Arc<T>` is an antipattern if the Arc isn't needed, it's simpler and clearer to just take in the ref directly
```suggestion
    rpc_client: &RpcClient,
```